probe/probe.go

```diff
@@ -360,119 +379,197 @@ type KubeArmorProbeData struct {
 	HostVisibility          string
 }
 
-func isKubeArmorRunning(c *k8s.Client, o Options) bool {
-	_, err := getKubeArmorDaemonset(c, o)
-	return err == nil
+func isKubeArmorRunning(c *k8s.Client, o Options) (bool, *Status) {
+	isRunning, DaemonsetStatus := getKubeArmorDaemonset(c, o)
+	return isRunning, DaemonsetStatus
+
+}
 
+// Status data
+type Status struct {
+	Desired   string `json:"desired"`
+	Ready     string `json:"ready"`
+	Available string `json:"available"`
 }
 
-func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, error) {
+func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, *Status) {
 	var data [][]string
 	// KubeArmor DaemonSet
 	w, err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Get(context.Background(), "kubearmor", metav1.GetOptions{})
 	if err != nil {
-		return false, err
+		return false, nil
 	}
-	color.Green("\nFound KubeArmor running in Kubernetes\n\n")
-	_, err = boldWhite.Printf("Daemonset :\n")
-	if err != nil {
-		color.Red(" Error while printing")
+	if o.Output == "text" {
+		color.Green("\nFound KubeArmor running in Kubernetes\n\n")
+		_, err = boldWhite.Printf("Daemonset :\n")
+		if err != nil {
+			color.Red(" Error while printing")
+		}
 	}
+
 	desired, ready, available := w.Status.DesiredNumberScheduled, w.Status.NumberReady, w.Status.NumberAvailable
 	if desired != ready && desired != available {
+
 		data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
 		return false, nil
 	}
+	if o.Output == "json" {
+		DaemonSetStatus := Status{
+			Desired:   strconv.Itoa(int(desired)),
+			Ready:     strconv.Itoa(int(ready)),
+			Available: strconv.Itoa(int(available)),
+		}
+		return true, &DaemonSetStatus
+	}
 	data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
 	renderOutputInTableWithNoBorders(data)
 	return true, nil
 }
-func getKubeArmorDeployments(c *k8s.Client, o Options) {
-	var data [][]string
-	_, err := boldWhite.Printf("Deployments : \n")
-	if err != nil {
-		color.Red(" Error while printing")
+
+func getKubeArmorDeployments(c *k8s.Client, o Options) map[string]*Status {
+
+	if o.Output == "text" {
+		_, err := boldWhite.Printf("Deployments : \n")
+		if err != nil {
+			color.Red(" Error while printing")
+		}
 	}
 	kubearmorDeployments, err := c.K8sClientset.AppsV1().Deployments(o.Namespace).List(context.Background(), metav1.ListOptions{
 		LabelSelector: "kubearmor-app",
 	})
-
 	if err != nil {
-		return
+		return nil
```

Comment:
why are we sending nil if there is an error?

Reply:
the nil is for empty `Status` structure, if  `err!=nil`